server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -138,6 +216,17 @@ private void validate(SetRequest request, Optional<ComponentDto> component) {
     checkComponentQualifier(request, definition, component);
   }
 
+  private void checkFieldType(SetRequest request, @Nullable PropertyDefinition definition, ListMultimap<String, String> valuesByFieldKeys) {
+    for (PropertyFieldDefinition fieldDefinition : definition.fields()) {
+      for (String value : valuesByFieldKeys.get(fieldDefinition.key())) {
+        PropertyDefinition.Result result = fieldDefinition.validate(value);
+        checkRequest(result.isValid(),
+          i18n.message(Locale.ENGLISH, "property.error." + result.getErrorKey(), "Error when validating setting with key '%s'. Field '%s' has incorrect value '%s'."),
```

Comment:
The message "Error when ..." is never displayed when bundle are not mocked (as in your UT).
For instance, here's the message I got : `` "Valid options are \"true\" and \"false\"`` => no idea which field was wrong.
I would suggest to not use i18n message when validating property set, and to always display your message "Error when...".